src/coll/options.rs

```diff
@@ -74,12 +74,20 @@ pub enum Hint {
 }
 
 impl Hint {
+    #[cfg(test)]
     pub(crate) fn to_bson(&self) -> Bson {
         match self {
             Hint::Keys(ref d) => Bson::Document(d.clone()),
             Hint::Name(ref s) => Bson::String(s.clone()),
         }
     }
+
+    pub(crate) fn to_raw_bson(&self) -> Result<RawBson> {
```

Comment:
It seems like this could just be `to_bson().try_into()?` ?

Reply:
I added this as a minor optimization to avoid an intermediary allocation of a regular `Bson` via `to_bson`. It did seem unnecessary to have both `to_bson` and `to_raw_bson` though so I just refactored the few uses of `to_bson` and removed it.

Ideally we'd have a `to_raw_bson(value: impl Serialize)` method in bson, but I'm not sure how that would actually work; the raw serializer spits out raw document bytes as opposed to the non-raw serializer, which generates BSON values.